v_4_0_0/filemanager.go

```diff
@@ -46,3 +49,22 @@ func RenderFiles(filesdir string, ctx interface{}) (Files, error) {
 
 	return files, nil
 }
+
+// GetIgnitionPath returns path for the ignition assets based on
+// base ignition directory and package subdirectory with assets.
+func GetIgnitionPath(ignitionDir string) string {
+	return filepath.Join(ignitionDir, FilesDir)
+}
+
+// GetPackagePath returns top package path for the current runtime file.
+// For example, for /go/src/k8scloudconfig/v_4_0_0/file.go function
+// returns /go/src/k8scloudconfig.
+// This function used only in tests for retrieving ignition assets in runtime.
+func GetPackagePath() (string, error) {
+	_, filename, _, ok := runtime.Caller(0)
+	if !ok {
+		return "", microerror.New("failed to retrieve runtime information")
+	}
+
+	return filepath.Dir(filepath.Dir(filename)), nil
+}
```

Comment:
In general I'm don't get how the whole ignition template is going to be generated. How is the `Files` type used and where?

Reply:
From our chat: they are just in the main struct and iterated and referenced in the template.